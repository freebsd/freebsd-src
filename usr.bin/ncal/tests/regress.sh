
CAL_BIN="ncal"
CAL="${CAL_BIN} -C"
NCAL="${CAL_BIN} -N"
YEARS="2008 2009 2010 2011"
ONEYEAR="2009"

echo 1..89

REGRESSION_START($1)

#
# The first tests are layout tests, to make sure that the output is still the
# same despite varying months.
#

# Full year calendars

for y in ${YEARS}; do
	# Regular calendar, Month days, No-highlight
	REGRESSION_TEST(`r-y${y}-md-nhl', `$NCAL -h ${y}')
	# Backwards calendar, Month days, No-highlight
	REGRESSION_TEST(`b-y${y}-md-nhl', `$CAL -h ${y}')
	# Regular calendar, Julian days, No-highlight
	REGRESSION_TEST(`r-y${y}-jd-nhl', `$NCAL -jh ${y}')
	# Backwards calendar, Julian days, No-highlight
	REGRESSION_TEST(`b-y${y}-jd-nhl', `$CAL -jh ${y}')
done

# 3 month calendars

for m in $(jot -w %02d 12); do
	# Regular calendar, Month days, No-highlight
	REGRESSION_TEST(`r-3m${ONEYEAR}${m}-md-nhl',
	    `$NCAL -h3 ${m} ${ONEYEAR}')
	# Backwards calendar, Month days, No-highlight
	REGRESSION_TEST(`b-3m${ONEYEAR}${m}-md-nhl', `$CAL -h3 ${m} ${ONEYEAR}')
	# Regular calendar, Julian days, No-highlight
	REGRESSION_TEST(`r-3m${ONEYEAR}${m}-jd-nhl',
	    `$NCAL -jh3 ${m} ${ONEYEAR}')
	# Backwards calendar, Julian days, No-highlight
	REGRESSION_TEST(`b-3m${ONEYEAR}${m}-jd-nhl', `$CAL -jh3 ${m} ${ONEYEAR}')
done

#
# The next tests are combinations of the various arguments.
#

# These should fail
REGRESSION_TEST(`f-3y-nhl',  `$NCAL -3 -y 2>&1')
REGRESSION_TEST(`f-3A-nhl',  `$NCAL -3 -A 3 2>&1')
REGRESSION_TEST(`f-3B-nhl',  `$NCAL -3 -B 3 2>&1')
REGRESSION_TEST(`f-3gy-nhl', `$NCAL -3 2008 2>&1')
REGRESSION_TEST(`f-3AB-nhl', `$NCAL -3 -A 3 -B 3 2>&1')
REGRESSION_TEST(`f-mgm-nhl', `$NCAL -m 3 2 2008 2>&1')
REGRESSION_TEST(`f-ym-nhl',  `$NCAL -y -m 2 2>&1')
REGRESSION_TEST(`f-ygm-nhl', `$NCAL -y 2 2008 2>&1')
REGRESSION_TEST(`f-yA-nhl',  `$NCAL -y -A 3 2>&1')
REGRESSION_TEST(`f-yB-nhl',  `$NCAL -y -B 3 2>&1')
REGRESSION_TEST(`f-yAB-nhl', `$NCAL -y -A 3 -B 3 2>&1')

# These should be successful

REGRESSION_TEST(`s-b-3-nhl',    `$CAL -d 2008.03 -3')
REGRESSION_TEST(`s-b-A-nhl',    `$CAL -d 2008.03 -A 1')
REGRESSION_TEST(`s-b-B-nhl',    `$CAL -d 2008.03 -B 1')
REGRESSION_TEST(`s-b-AB-nhl',   `$CAL -d 2008.03 -A 1 -B 1')
REGRESSION_TEST(`s-b-m-nhl',    `$CAL -d 2008.03 -m 1')
REGRESSION_TEST(`s-b-mgy-nhl',  `$CAL -d 2008.03 -m 1 2007')
REGRESSION_TEST(`s-b-gmgy-nhl', `$CAL -d 2008.03 1 2007')
REGRESSION_TEST(`s-r-3-nhl',    `$NCAL -d 2008.03 -3')
REGRESSION_TEST(`s-r-A-nhl',    `$NCAL -d 2008.03 -A 1')
REGRESSION_TEST(`s-r-B-nhl',    `$NCAL -d 2008.03 -B 1')
REGRESSION_TEST(`s-r-AB-nhl',   `$NCAL -d 2008.03 -A 1 -B 1')
REGRESSION_TEST(`s-r-m-nhl',    `$NCAL -d 2008.03 -m 1')
REGRESSION_TEST(`s-r-mgy-nhl',  `$NCAL -d 2008.03 -m 1 2007')
REGRESSION_TEST(`s-r-gmgy-nhl', `$NCAL -d 2008.03 1 2007')

REGRESSION_END()
