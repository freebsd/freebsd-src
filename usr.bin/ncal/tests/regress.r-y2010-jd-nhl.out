                                  2010
    January                 February                March                   
Mo       4  11  18  25      32  39  46  53          60  67  74  81  88    
Tu       5  12  19  26      33  40  47  54          61  68  75  82  89    
We       6  13  20  27      34  41  48  55          62  69  76  83  90    
Th       7  14  21  28      35  42  49  56          63  70  77  84        
Fr   1   8  15  22  29      36  43  50  57          64  71  78  85        
Sa   2   9  16  23  30      37  44  51  58          65  72  79  86        
Su   3  10  17  24  31      38  45  52  59          66  73  80  87        

    April                   May                     June                    
Mo      95 102 109 116         123 130 137 144 151     158 165 172 179    
Tu      96 103 110 117         124 131 138 145     152 159 166 173 180    
We      97 104 111 118         125 132 139 146     153 160 167 174 181    
Th  91  98 105 112 119         126 133 140 147     154 161 168 175        
Fr  92  99 106 113 120         127 134 141 148     155 162 169 176        
Sa  93 100 107 114         121 128 135 142 149     156 163 170 177        
Su  94 101 108 115         122 129 136 143 150     157 164 171 178        

    July                    August                  September               
Mo     186 193 200 207         214 221 228 235 242     249 256 263 270    
Tu     187 194 201 208         215 222 229 236 243     250 257 264 271    
We     188 195 202 209         216 223 230 237     244 251 258 265 272    
Th 182 189 196 203 210         217 224 231 238     245 252 259 266 273    
Fr 183 190 197 204 211         218 225 232 239     246 253 260 267        
Sa 184 191 198 205 212         219 226 233 240     247 254 261 268        
Su 185 192 199 206         213 220 227 234 241     248 255 262 269        

    October                 November                December                
Mo     277 284 291 298     305 312 319 326 333         340 347 354 361    
Tu     278 285 292 299     306 313 320 327 334         341 348 355 362    
We     279 286 293 300     307 314 321 328         335 342 349 356 363    
Th     280 287 294 301     308 315 322 329         336 343 350 357 364    
Fr 274 281 288 295 302     309 316 323 330         337 344 351 358 365    
Sa 275 282 289 296 303     310 317 324 331         338 345 352 359        
Su 276 283 290 297 304     311 318 325 332         339 346 353 360        
